package.json

```diff
@@ -14,12 +14,15 @@
     "url": "https://github.com/iov-one/ethereum-smart-contracts/issues"
   },
   "scripts": {
-    "compile": "truffle compile",
+    "flatten": "rm -rf ./build/flattened && ./scripts/flatten.sh",
+    "prebuild": "yarn flatten",
+    "build": "truffle compile",
     "format": "prettier --write --loglevel warn \"./{,migrations,test}/*.{js,json,md}\"",
     "lint": "yarn run lint:js && yarn run lint:sol",
     "lint:js": "eslint .",
     "lint:js:fix": "eslint --fix .",
     "lint:sol": "solhint \"contracts/**/*.sol\"",
+    "pretest": "yarn flatten",
```

Comment:
Cool, I didn't know about pretest. I like it.
Can I do preX for anything, or just build, test, deploy?

Reply:
During development, I also had "precompile" and it worked for "compile". So I guess it works for everything, but I don't know.

If preX works, then prepreX and preprepreX could also work recursively …

Reply:
okay, it works for every X, but only one level deep.

```

    "prepreprefoobar": "echo 1",
    "preprefoobar": "echo 2",
    "prefoobar": "echo 3",
    "foobar": "echo 4",
```

prints 3, 4